packages/perseus-editor/src/widgets/__tests__/radio-editor.test.tsx

```diff
@@ -176,4 +176,61 @@ describe("radio-editor", () => {
 
         expect(options?.numCorrect).toEqual(2);
     });
+
+    it("derives num correct when calling onChange", async () => {
+        let options: any;
+
+        function getCorrectChoice(): PerseusRadioChoice {
+            return {
+                content: "",
+                correct: true,
+            };
+        }
+
+        function getIncorrectChoice(): PerseusRadioChoice {
+            const choice = getCorrectChoice();
+            choice.correct = false;
+            return choice;
+        }
+
+        render(
+            <RadioEditor
+                onChange={(o) => {
+                    options = o;
+                }}
+                apiOptions={ApiOptions.defaults}
+                static={false}
+                multipleSelect={true}
+                countChoices={true}
+                choices={[
+                    // start with 2 correct, 2 incorrect
+                    getIncorrectChoice(),
+                    getIncorrectChoice(),
+                    getCorrectChoice(),
+                    getCorrectChoice(),
+                ]}
+            />,
+            {wrapper: RenderStateRoot},
+        );
+
+        /**
+         * This was super annoying to figure out.
+         * When in "edit" mode (which the editor is obviously)
+         * the only way to select an option from the radio
+         * is to click the A/B/C/etc icons themselves.
+         * We have a `elem.getAttribute("data-is-radio-icon")` check
+         * to make sure that what we're clicking is the icon and
+         * not some other part of the radio choice.
+         * It's just a div, hence the test ID.
+         */
+        const choices = screen.getAllByTestId(
+            "choice-icon__library-choice-icon",
+        );
+
+        // switch an incorrect answer into a correct answer
+        await userEvent.click(choices[0]);
+
+        // now there should be 3 correct answers
+        expect(options?.numCorrect).toBe(3);
```

Comment:
Stylistically, I find using an `onChange` `jest.fn()` for these types of things simpler to grok. With a var we set in the callback you have a to trace up through the test to understand that the `onChange` callback sets it. 

With an `onChange` mock, we could use:

```ts
expect(onChange).toHaveBeenCalledWith(expect.objectContaining({
    numCorrect: 3
});
```

In my mind that's less logic to support the test than having a `let options` we manage. 🤷‍♂️

Reply:
Not a hill I'll die on, but I'm not totally sure I agree. I think...

``` TS
expect(options?.numCorrect).toBe(3);
```

...is much more legible than...

``` TS
expect(onChangeMock).toHaveBeenCalledWith(
    expect.objectContaining({
        numCorrect: 3,
    }),
);
```

I also think from a KISS perspective that a simple callback is easier to follow/debug than a Jest abstraction. But it's not a big deal in this particular case, so I made the change.